HPXMLtoOpenStudio/resources/hpxml_schematron/EPvalidator.xml

```diff
@@ -1314,6 +1333,34 @@
+      <sch:assert role='ERROR' test='number(h:BoreholesOrTrenches/h:Count) &lt;= 10 or not(h:BoreholesOrTrenches/h:Count)'>Expected BoreholesOrTrenches/Count to be less than or equal to 10</sch:assert>
+      <sch:assert role='ERROR' test='count(h:BoreholesOrTrenches/h:Length) &lt;= 1'>Expected 0 or 1 element(s) for xpath: BoreholesOrTrenches/Length</sch:assert>      
+      <sch:assert role='ERROR' test='number(h:BoreholesOrTrenches/h:Length) &gt;= 79 or not(h:BoreholesOrTrenches/h:Length)'>Expected BoreholesOrTrenches/Length to be greater than or equal to 79</sch:assert>
+      <sch:assert role='ERROR' test='number(h:BoreholesOrTrenches/h:Length) &lt;= 500 or not(h:BoreholesOrTrenches/h:Length)'>Expected BoreholesOrTrenches/Length to be less than or equal to 500</sch:assert>
+      <sch:assert role='ERROR' test='count(h:BoreholesOrTrenches/h:Spacing) &lt;= 1'>Expected 0 or 1 element(s) for xpath: BoreholesOrTrenches/Spacing</sch:assert>
+      <sch:assert role='ERROR' test='number(h:BoreholesOrTrenches/h:Spacing) &gt; 0 or not(h:BoreholesOrTrenches/h:Spacing)'>Expected BoreholesOrTrenches/Spacing to be greater than 0</sch:assert>
+      <sch:assert role='ERROR' test='count(h:BoreholesOrTrenches/h:Diameter) &lt;= 1'>Expected 0 or 1 element(s) for xpath: BoreholesOrTrenches/Diameter</sch:assert>
+      <sch:assert role='ERROR' test='number(h:BoreholesOrTrenches/h:Diameter) &gt; 0 or not(h:BoreholesOrTrenches/h:Diameter)'>Expected BoreholesOrTrenches/Diameter to be greater than 0</sch:assert>
+      <sch:assert role='ERROR' test='count(h:Grout/h:Type) + count(h:Grout/h:Conductivity) &gt;= 0'>Expected 0 or more element(s) for xpath: Grout/Type | Grout/Conductivity</sch:assert>
+      <sch:assert role='ERROR' test='count(h:Pipe/h:Type) + count(h:Pipe/h:Conductivity) &gt;= 0'>Expected 0 or more element(s) for xpath: Pipe/Type | Pipe/Conductivity</sch:assert>
+      <sch:assert role='ERROR' test='count(h:Pipe/h:Diameter) &lt;= 1'>Expected 0 or 1 element(s) for xpath: Pipe/Diameter</sch:assert>
+      <sch:assert role='ERROR' test='number(h:Pipe/h:Diameter) = 0.75 or number(h:Pipe/h:Diameter) = 1.0 or number(h:Pipe/h:Diameter) = 1.25 or not(h:Pipe/h:Diameter)'>Expected Pipe/Diameter to be 0.75, 1.0, or 1.25</sch:assert>
+      <sch:assert role='ERROR' test='count(h:Pipe/h:ShankSpacing) &lt;= 1'>Expected 0 or 1 element(s) for xpath: Pipe/ShankSpacing</sch:assert>
+      <sch:assert role='ERROR' test='number(h:Pipe/h:ShankSpacing) &gt; 0 or not(h:Pipe/h:ShankSpacing)'>Expected Pipe/ShankSpacing to be greater than 0</sch:assert>
+      <sch:assert role='ERROR' test='count(h:BoreholesOrTrenches/h:Count) + count(h:extension/h:BorefieldConfiguration) = 2 or not(h:extension/h:BorefieldConfiguration)'>Expected 0 or 2 element(s) for xpath: BoreholesOrTrenches/Count | extension/BorefieldConfiguration</sch:assert>
```

Comment:
Similar to previous comment... Why do we need both? If they provide just Count, can we not choose an appropriate configuration (and vice versa)?

Reply:
We support a provided Count and defaulted configuration (always rectangle). Maybe my error message can be improved? If we supported a provided configuration and defaulted Count, seems like you could run into trouble (i.e., no corresponding g-function values).

Reply:
How about we meet somewhere in the middle? I updated the schematron to allow defaulting of Count if (and only if) Rectangle is the provided configuration (it has corresponding g-function values for Count that is 1 - 10). I added a schematron validation test for checking that any other configuration type throws an error. I added a defaults test for checking that defaulted Count and provided Rectangle configuration is actually allowed. I updated the docs to call out this nuance.

Reply:
Is there a technical reason we can't have a user tell us the borefield configuration and choose an autosized number of boreholes that is valid for that configuration?

Reply:
Maybe I don't understand what you mean by "choose". Autosized number of boreholes is based on `Cool_Capacity`: https://github.com/NREL/OpenStudio-HPXML/blob/geothermal_loop/HPXMLtoOpenStudio/resources/hvac_sizing.rb#L1948. Maybe you're saying iteratively increase/decrease the original number of boreholes (while simultaneously adjusting the depth) so that we can possibly arrive at a valid number of boreholes?

Reply:
Basically, yes. I wouldn't think any iteration is needed -- you can just choose the closest number of boreholes to what was autosized, no?

Reply:
Well, but then you could possibly exceed our maximum prescribed bore depth, no? E.g., for "U" you autosize to 8 boreholes. This is not valid, so you try 7. This causes, based on the total borehole length, the average bore depth to exceed 500 ft. So instead of 7, 9 would've worked...

This seems overly complicated for how uncommon some of these configurations likely are.

Reply:
Yeah I agree with Joe here that I think this is a pretty niche use case where someone provides a configuration and we have to autosize the # of boreholes.

The one technical way you could get into trouble I think is for buildings with very small loads (say a unit in a MF building). Some of the configuration like open rectangle require a minimum of 8 boreholes, if loads are small enough you'll start getting close to the minimum depth where you'd need to extrapolate. That's an edge case for sure as well, but seems to be the only time you'd really run into a case where we'd have to throw an error.